README.tfdoc.hcl

```
@@ -0,0 +1,531 @@
+            END
+          }
+        }
+
+        variable "view" {
+          type        = any
+          readme_type = "object(view)"
+          default     = []
+          description = <<-END
+            A view from a different dataset to grant access to.
+          END
+
+          attribute "project_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the project containing this table.
+            END
+          }
+
+          attribute "table_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the dataset containing this table.
+            END
+          }
+
+          attribute "dataset_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              The ID of the table.
+            END
+          }
+        }
+
+        variable "role" {
+          type = any
+          readme_type = <<-END
+            list(object({
+            role           = string
+            special_group  = string
+            group_by_email = string
+            user_by_email  = string
+          }))
+          END
+          default     = []
+          description = <<-END
+            (Optional) A map of dataset-level roles including the role, special_group, group_by_email, and user_by_email
+          END
+        }
+
+        variable "default_encryption_configuration" {
+          type        = any
+          readme_type = "object(default_encryption_configuration)"
+          description = <<-END
+            The default encryption key for all tables in the dataset. Once this property is set, all newly-created partitioned tables in the dataset will have encryption key set to this value, unless table creation request (or query) overrides the key.
+          END
+
+          attribute "kms_key_name" {
+            required    = true
+            type        = string
+            description = <<-END
+              Describes the Cloud KMS encryption key that will be used to protect destination BigQuery table.
+              The BigQuery Service Account associated with your project requires access to this encryption key.
+            END
+          }
+        }
+
+        variable "delete_contents_on_destroy" {
+          type        = bool
+          default     = false
+          description = <<-END
+            If set to true, delete all the tables in the dataset when destroying the resource; otherwise, destroying the resource will fail if tables are present.
+          END
+        }
+
+        variable "authoritative" {
+          type        = bool
+          default     = true
+          description = <<-END
+            (Optional) google bigquery dataset iam choice whether 'authoritative == true' use google_bigquery_iam_binding, 'authoritative == false' use google_bigquery_iam_member
```

this text is outdated... please replace whenever you see this.

```suggestion
            Whether to exclusively set (authoritative mode) or add (non-authoritative/additive mode) members to the role.
```